File: 22.链表中倒数第k个结点/22.链表中倒数第k个结点.py
class ListNode:
    def __init__(self,x):
        self.val = x
        self.next = None
class Solution:
    def FindKthToTail(self, head, k):
        if head == None or k <1:
            return None
        p = head
        q = head      
        step = 1
        while p.next is not None:
            if step >= k and q is not None:
                q = q.next
            p = p.next
            step += 1

        if k <= step:
            return q.val
        else:
            return -1

File: 22.链表中倒数第k个结点/22.链表中倒数第k个结点.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* FindKthToTail(ListNode* pListHead, unsigned int k) {
      ListNode* fast = pListHead;
      ListNode* slow = pListHead;
      while(fast != nullptr && k!=0){
          fast = fast->next;
          k--;
      };
      if (k>0)
          return nullptr;
      while(fast != nullptr){
          fast = fast->next;
          slow = slow->next;
      };
      return slow;
    }
};